### ISSUE TITLE
lib path smafu (?)

### ISSUE THREAD

**Issue by gaborcsardi** [jimhester/covr#166]
I have a custom directory for user packages. When running `covr::package_coverage()`, I get:

```r
❯ covr::package_coverage()
Error in run_commands(pkg, tmp_lib, code) : Error running commands:

R version 3.2.4 (2016-03-10) -- "Very Secure Dishes"
Copyright (C) 2016 The R Foundation for Statistical Computing
Platform: x86_64-apple-darwin13.4.0 (64-bit)

R is free software and comes with ABSOLUTELY NO WARRANTY.
You are welcome to redistribute it under certain conditions.
Type 'license()' or 'licence()' for distribution details.

  Natural language support but running in an English locale

R is a collaborative project with many contributors.
Type 'contributors()' for more information and
'citation()' on how to cite R or R packages in publications.

Type 'demo()' for some demos, 'help()' for on-line help, or
'help.start()' for an HTML browser interface to help.
Type 'q()' to quit R.

> library('packageMetrics2')
Error : .onLoad failed in loadNamespace() for 'covr', details:
  call: loadNamespace(name)
  error: there is no package called ‘rex’
Error: package or namespace load failed for ‘packageMetrics2’
Execution halted
```
This is what I have:
```
❯ .libPaths()
[1] "/Users/gaborcsardi/r_pkgs"
[2] "/Library/Frameworks/R.framework/Versions/3.2/Resources/library"
```

The following patch fixes this, but it might break sg else. Nevertheless, I think it is better to use the user's `libPaths()`, not (just) `R_LIBS_USER`. Want a PR? :)

```diff
diff --git a/R/covr.R b/R/covr.R
index d9584a2..5a7616f 100644
--- a/R/covr.R
+++ b/R/covr.R
@@ -200,7 +200,7 @@ package_coverage <- function(path = ".",

   withr::with_envvar(c(
     SHLIB_LIBADD = "--coverage",
-      R_LIBS_USER = env_path(tmp_lib, Sys.getenv("R_LIBS_USER"))), {
+      R_LIBS_USER = env_path(tmp_lib, .libPaths())), {
     withr::with_libpaths(tmp_lib, action = "prefix", {
       withCallingHandlers({
       if ("vignettes" %in% type) {
```

**Comment by jimhester** [comment]
I agree, simply using the users current `.libPaths()` seems the most straightforward approach. I have a branch with this at https://github.com/jimhester/covr/tree/simplify_environments which I will probably merge in the near future after I have done some more testing.

**Comment by gaborcsardi** [comment]
Cool!
I found it rather difficult to exactly reproduce the library path in a "child" R session. FWIW this is what I came up with in the end, it is from https://github.com/MangoTheCat/rcmdcheck/blob/af79cc1d193972f318f3b80e9c15d0838280c48d/R/safe_check.R#L9-L13

```r
lib <- paste(.libPaths(), collapse = .Platform$path.sep)
profile <- tempfile()
with_envvar(
  c(R_LIBS = lib,
    R_LIBS_USER = lib,
    R_LIBS_SITE = lib,
    R_PROFILE_USER = profile),
  R(...)
)
```

It has not been tested much, though. 

Btw. this is something that our future `callr` package could do. :)
